daemon/src/model.rs

```diff
@@ -298,4 +522,55 @@ mod tests {
 
         assert!(past_event.has_likely_occured());
     }
+
+    #[test]
+    fn algebra_with_usd() {
+        let usd_0 = Usd::new(dec!(1.234));
+        let usd_1 = Usd::new(dec!(9.876));
+
+        let usd_sum = usd_0 + usd_1;
+        let usd_diff = usd_0 - usd_1;
+        let half = usd_0 / 2;
+        let double = usd_1 * 2;
+
+        assert_eq!(usd_sum.value, dec!(11.110));
+        assert_eq!(usd_diff.value, dec!(-8.642));
+        assert_eq!(half.value, dec!(0.617));
+        assert_eq!(double.value, dec!(19.752));
+    }
+
+    #[test]
+    fn algebra_with_exchange_rate() {
```

Comment:
This might be more expressive of what is happening here.

```suggestion
    fn usd_for_1_btc_buys_1_btc() {
```